ibm/resource_ibm_iam_account_settings.go

```diff
@@ -0,0 +1,393 @@
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"timestamp": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Timestamp when the action was triggered.",
+						},
+						"iam_id": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "IAM ID of the identity which triggered the action.",
+						},
+						"iam_id_account": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Account of the identity which triggered the action.",
+						},
+						"action": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Action of the history entry.",
+						},
+						"params": &schema.Schema{
+							Type:        schema.TypeList,
+							Required:    true,
+							Description: "Params of the history entry.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"message": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Message which summarizes the executed action.",
+						},
+					},
+				},
+			},
+			"session_expiration_in_seconds": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Defines the session expiration in seconds for the account. Valid values:  * Any whole number between between '900' and '86400'  * NOT_SET - To unset account setting and use service default.",
+			},
+			"session_invalidation_in_seconds": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Defines the period of time in seconds in which a session will be invalidated due  to inactivity. Valid values:   * Any whole number between '900' and '7200'   * NOT_SET - To unset account setting and use service default.",
+			},
+		},
+	}
+}
+
+func resourceIbmIamAccountSettingsCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamIdentityClient, err := meta.(ClientSession).IamIdentityV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getAccountSettingsOptions := &iamidentityv1.GetAccountSettingsOptions{}
+
+	userDetails, err := meta.(ClientSession).BluemixUserDetails()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	getAccountSettingsOptions.SetAccountID(userDetails.userAccount)
+	if _, ok := d.GetOk("include_history"); ok {
+		getAccountSettingsOptions.SetIncludeHistory(d.Get("include_history").(bool))
+	}
+
+	accountSettingsResponse, response, err := iamIdentityClient.GetAccountSettingsWithContext(context, getAccountSettingsOptions)
+	if err != nil {
+		log.Printf("[DEBUG] GetAccountSettingsWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", *getAccountSettingsOptions.AccountID, *accountSettingsResponse.AccountID))
```

Comment:
why the ID  is combination of two accountsID 
wt is the difference b/w getAccountSettingsOptions and accountSettingsResponse accountId

Reply:
I have no idea why this is the case, this is how the code was generated. I did amend it to have a single account id as the ID, but it fails as it's expecting two values separated by a '/' for some reason. It works fine as is so I didn't delve into it much deeper than that.

Reply:
I've fixed this now to only take the single account id.